### ISSUE TITLE
Spread in object literals hides rules violation

### ISSUE THREAD

**Issue by sometimesalready** [eslint/eslint#11315]
**Tell us about your environment**

* **ESLint Version:** 5.12.1

**What parser (default, Babel-ESLint, etc.) are you using?** babel-eslint

```js
const {param1, param2, ...restParams} = params;
console.log(param1)
// console.log(param2)
```

**What did you expect to happen?**
eslint show some rule violations. in our example `[no-unused-vars]` for `param2`

**What actually happened? Please include the actual, raw output from ESLint.**
eslint do not show any rule violations if **there is spred in object literals**

**Are you willing to submit a pull request to fix this bug?**
no

**Comment by platinumazure** [comment]
Hi @sometimesalready, could you please share your configuration?

The rule has an [`ignoreRestSiblings` option](https://eslint.org/docs/rules/no-unused-vars#ignorerestsiblings) which controls this behavior. By default, it is false and variables should be reported; but maybe your configuration has enabled this option. If your configuration has the option enabled, then the rule is working as designed.

**Comment by sometimesalready** [comment]
Hi @platinumazure, yes - you are right: `ignoreRestSiblings` option was set to true. Thank you!
